include/highfive/bits/H5Easy_misc.hpp

```diff
@@ -0,0 +1,97 @@
+/*
+ *  Copyright (c), 2017, Adrien Devresse <adrien.devresse@epfl.ch>
+ *
+ *  Distributed under the Boost Software License, Version 1.0.
+ *    (See accompanying file LICENSE_1_0.txt or copy at
+ *          http://www.boost.org/LICENSE_1_0.txt)
+ *
+ */
+#ifndef H5EASY_BITS_MISC_HPP
+#define H5EASY_BITS_MISC_HPP
+
+#include "../H5Easy.hpp"
+
+namespace HighFive
+{
+    namespace detail
+    {
+        inline auto error(const HighFive::File& file, const std::string& path, std::string message)
```

Comment:
Thanks, much better, I was not aware! I think that in the spirit of `HighFive::Exception` it would be best to return a `HighFive::Exception` and throw where invoked. This is what I have now done.